server/api/utils/singletons/k8s.py

```diff
@@ -533,7 +533,7 @@ def store_secrets(
                     )
                 raise exc
 
-        secret_data = k8s_secret.data.copy()
+        secret_data = k8s_secret.data.copy() if k8s_secret.data else {}
         for key, value in secrets.items():
```

Comment:
If the `secrets` parameter is `None`, the code will raise an error when attempting to iterate over `secrets.items()`